core/src/main/java/org/axonframework/serialization/AbstractXStreamSerializer.java

```diff
@@ -145,6 +146,16 @@ protected AbstractXStreamSerializer(Charset charset, XStream xStream, RevisionRe
 
         xStream.alias("meta-data", MetaData.class);
         xStream.registerConverter(new MetaDataConverter(xStream.getMapper()));
+
+        // Make sure ClassLoader of this instance is registered - required by spring devtools RestartClassLoader
+        if (CompositeClassLoader.class.isAssignableFrom(xStream.getClassLoader().getClass())) {
```

Comment:
You can actually use `CompositeClassLoader.class.isInstance(xStream.getClassLoader())` here.